datafusion/core/src/datasource/listing/table.rs

```diff
@@ -2096,4 +2230,64 @@ mod tests {
         // Return Ok if the function
         Ok(())
     }
+
+    /// tests insert into with end to end sql
+    /// create external table + insert into statements
+    async fn helper_test_insert_into_sql(
```

Comment:
I think we could use sqllogictest to test this as well -- 

Perhaps add to https://github.com/apache/arrow-datafusion/blob/main/datafusion/core/tests/sqllogictests/test_files/insert.slt

I can't remember how temporary directories work in sqllogictests though

Reply:
I actually worked on this in #7283 which I just opened for copy.slt. Once those updates are in, I can cut another PR to add additional tests to `insert.slt`